datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -1991,6 +1992,71 @@ pub fn array_intersect(args: &[ArrayRef]) -> Result<ArrayRef> {
     }
 }
 
+pub fn general_array_distinct<OffsetSize: OffsetSizeTrait>(
+    array: &GenericListArray<OffsetSize>,
+    field: &FieldRef,
+) -> Result<ArrayRef> {
+    let dt = array.value_type();
+    let mut offsets = vec![OffsetSize::usize_as(0)];
+    let mut new_arrays = vec![];
+    let converter = RowConverter::new(vec![SortField::new(dt.clone())])?;
+    // distinct for each list in ListArray
+    for arr in array.iter().flatten() {
+        let values = converter.convert_columns(&[arr])?;
+        let mut rows = Vec::with_capacity(values.num_rows());
+        // sort elements in list and remove duplicates
+        for val in values.iter().sorted().dedup() {
+            rows.push(val);
+        }
```

Comment:
It seems that using `collect::<Vec<_>>()` can avoid waste memory in vector

Reply:
the vector's capacity is initialized by `values.num_rows()`, but if `values` has a lot duplicated element, the vector may only use part of it's memory.

Reply:
Thanks, get it!